engine/classes/Elgg/Database/AdminNotices.php

```diff
@@ -61,27 +65,26 @@ function add($id, $message) {
 	 *
 	 * @return bool
 	 */
-	function delete($id) {
-		if (!$id) {
-			return false;
-		}
-		
+	public function delete($id = '') {
 		$result = true;
 		
-		$notices = elgg_get_entities([
-			'type' => 'object',
-			'subtype' => 'admin_notice',
+		$notices = $this->find([
 			'metadata_name' => 'admin_notice_id',
 			'metadata_value' => $id,
```

Comment:
maybe you could also combine these in a metadata_name_value_pairs? I dislike the separate ones

Reply:
Maybe in some other lifetime

Reply:
well this is not what you actually want... plus it is a more complicated query... as it creates an extra join if i am correct

Reply:
QueryBuilder is smart enough to determine the correct query to use based on number of metadata wheres

Reply:
can you show me where??

Reply:
https://github.com/Elgg/Elgg/blob/master/engine/classes/Elgg/Database/Entities.php#L255

Reply:
sweet...